packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,74 @@
-    //   do with this.
-    // remove_experiment_parts(experiment_id).or_else(|e| {
-    //     let err_string = e.to_string();
-    //     if err_string.contains("kind: NotFound") {
-    //         Ok(())
-    //     } else {
-    //         Err(Error::from(format!(
-    //             "Error while trying to remove experiment parts folder: {err_string}"
-    //         )))
-    //     }
-    // })?;
-
-    Ok(())
+pub fn cleanup_experiment(experiment_id: &ExperimentId, exit_status: EngineExitStatus) {
+    MemoryId::clean_up(experiment_id);
+
+    // Cleanup python socket files in case the engine didn't
+    let frompy_files = glob::glob(&format!("{experiment_id}-frompy*"));
+    let topy_files = glob::glob(&format!("{experiment_id}-topy*"));
+
+    frompy_files
+        .into_iter()
+        .chain(topy_files)
+        .flatten()
+        .flatten()
+        .for_each(|path| match std::fs::remove_file(&path) {
+            Ok(_) => {
+                match exit_status {
+                    EngineExitStatus::Success => tracing::error!(
+                        experiment = %experiment_id,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                    EngineExitStatus::Error => tracing::warn!(
+                        experiment = %experiment_id,
+                        "Removed file {path:?} that should've been cleanup by the engine."
+                    ),
+                };
+            }
+            Err(err) => {
+                tracing::warn!(
+                    experiment = %experiment_id,
+                    "Could not clean up {path:?}: {err}"
+                );
+            }
+        });
+
+    remove_experiment_parts(experiment_id);
 }
 
-#[allow(dead_code)]
-fn remove_experiment_parts(experiment_id: &ExperimentId) -> Result<()> {
-    let mut base_path = PathBuf::from(RELATIVE_PARTS_FOLDER);
-    // TODO: this is unused at the moment so it's fine, but this logic is wrong, we name our folders
-    //  differently, we should update the design to store the paths and use them here when we use
-    //  the clean up code again
-    base_path.push(experiment_id.to_string());
-    tracing::trace!("Removing all parts files in: {base_path:?}");
-    std::fs::remove_dir_all(base_path)?;
-    Ok(())
+fn remove_experiment_parts(experiment_id: &ExperimentId) {
+    let path = format!("{RELATIVE_PARTS_FOLDER}/{experiment_id}");
+    match std::fs::remove_dir_all(&path) {
+        Ok(_) => {
+            tracing::trace!(
+                experiment = %experiment_id,
+                "Removed parts folder for experiment {experiment_id}: {path:?}"
+            );
+        }
+        Err(err) => {
+            tracing::warn!(
+                experiment = %experiment_id,
+                "Could not clean up {path:?}: {err}"
+            );
+        }
+    }
+
+    match std::fs::remove_dir(RELATIVE_PARTS_FOLDER) {
```

Comment:
We shouldn't be doing this, because another experiment might be running in which case we delete the parts during execution...

Reply:
`remove_dir` will only do something if the folder is empty. If another experiment is running it shouldn't be.

Reply:
`remove_dir` is basically `rmdir`, `remove_dir_all` is basically the same as `rm -r`.

Reply:
At the very least there should be a comment explaining that. I still worry this might lead to some obscure race condition in the future where an engine might create the parts folder, another deletes it, and then the first one tries writing to it.

Reply:
The race condition is a good point, let's remove this then. We might want to change `./parts` to something in `/tmp` eventually, later on, to avoid having the `./parts` folder living in the engine folder.

Reply:
Removed it in https://github.com/hashintel/hash/pull/494